invokeai/app/services/model_install/model_install_base.py

```diff
@@ -103,19 +136,28 @@ class URLModelSource(StringLikeSource):
 
     url: AnyHttpUrl
     access_token: Optional[str] = None
-    type: Literal["generic_url"] = "generic_url"
+    type: Literal["url"] = "url"
 
     def __str__(self) -> str:
         """Return string version of the url when string rep needed."""
         return str(self.url)
 
+    @field_validator("url", mode="before")
+    @classmethod
+    def to_url(cls, raw: str) -> AnyHttpUrl:
+        """Change raw string into a Url."""
+        return AnyHttpUrl(raw)
```

Comment:
I think the idea was to allow caller to pass a string rather than a url without generating a runtime error, but this is just bad practice and I've deleted.